[Lee Sungmin] iP

woodenclock opened the pull request.

woodenclock committed:
Update Duke.java for Level 0

woodenclock committed:
Update Duke.java for Level 1

woodenclock committed:
Update Duke.java for Level 2

woodenclock committed:
Update Duke.java for Level 2

woodenclock committed:
Update Duke.java for Level 3

woodenclock committed:
Update Logo

woodenclock committed:
Update Duke.java for A-CodingStandard
Fix code such that when empty input keyed in, will recognize as invalid input

woodenclock committed:
Update ip

woodenclock committed:
Update Duke.java to Elvis.java Change in chatbot name

woodenclock commented on src/Task.java:
```diff
@@ -0,0 +1,100 @@
+    public static void manageTask() {
+        Scanner in = new Scanner(System.in);    //Scanner for standard input
+        Elvis.bootUp();
+        while (true) {
+            String inputBuffer = in.nextLine(); //Scans for the input
+            Scanner bufferScanner = new Scanner(inputBuffer);   //All inputs first stored in inputBuffer
+            if (!bufferScanner.hasNext()) {
+                System.out.println("Please input a valid input");
+                continue;
+            }
+            String firstWord = bufferScanner.next();
+            boolean hasInteger = bufferScanner.hasNextInt();
+            int numberInput = -1;
+            if (hasInteger) {
+                numberInput = bufferScanner.nextInt();
+            }
+
+            if (inputBuffer.contains("bye")) {
+                break;
+            } else if (inputBuffer.equals("list")) {
+                if (listCount == 0) {
+                    Elvis.printHorizontalLines();
+                    System.out.println("Nothing to list");
+                    Elvis.printHorizontalLines();
+                    continue;
+                }
+                Elvis.printHorizontalLines();
+                for (int i = 0; i < listCount; i++) {
+                    System.out.println(i + 1 + "." + "[" + list[i].getStatus() + "] " + list[i].getToDo());
+                }
+                Elvis.printHorizontalLines();
+            } else if (firstWord.equals("unmark") && hasInteger && !bufferScanner.hasNext()) {
+                int nthTask = numberInput - 1;
+                if (numberInput > listCount || nthTask < 0 || listCount == 0) {
+                    System.out.println("No such Task!");
```
Wow that's long

woodenclock committed:
Create Week 4 Level 4

woodenclock committed:
Update Elvis.java for Level 4

woodenclock committed:
Delete class files

woodenclock committed:
Update Elvis.java text-ui-test

woodenclock committed:
Update Elvis.java logo

ICubE01 commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# Elvis project template
```
I love your program's name :D

ICubE01 commented on src/Deadlines.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadlines extends Task {
+    private char taskType = 'D';
+    private String date;
+    public Deadlines(String description, String date) {
```
It would be good to insert blank line to separate variables and methods.

woodenclock replied:
wait... im confused.
I thought they were supposed to be not separated?
Aren't constructors considered a type of method?

![image](https://github.com/nus-cs2113-AY2324S1/ip/assets/69474977/e9f410b1-aaf7-453d-a73f-f0e4d2d5f772)

ICubE01 replied:
I mean, blank line between line 3 and line 4.
You did well, no space between method name and parenthesis.

Below is what I meant.
```
private String date;
// Blank line
public Deadlines(String description, String date) {
```

It's kinda related to code quality, not coding standard. Not necessary, but good for readability.

woodenclock replied:
Thanks!

ICubE01 commented on src/Echo.java:
I think this code isn't used. Please don't leave dummy code.

woodenclock replied:
Oh, this was for the previous ip levels haha

woodenclock replied:
Sorry, this is from the previous levels that I just left it in the repo

ICubE01 commented on src/Elvis.java:
```diff
@@ -0,0 +1,58 @@
+public class Elvis {
+    //Logo for ELVIS the Chatbot
+    public static String logo =
```
A constant variable name has to be uppercased.

ICubE01 commented on src/Deadlines.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadlines extends Task {
```
Class name would be good to written not in plural.

woodenclock replied:
I did this way, cuz i followed the ip instructions too closely?
i followed their name classes.
Is it better to have the class names in singular?


![image](https://github.com/nus-cs2113-AY2324S1/ip/assets/69474977/96f14b97-5987-4a63-8253-b320d73ba87b)

ICubE01 replied:
The class of your code is for single task, so the name of it should be singular.
If the class was related to multiple tasks, (e.g. list of tasks) then the name of class should be plural.
Similar as variable names.

ICubE01 commented on src/List.java:
I think this code isn't used.

woodenclock replied:
Sorry, this is from the previous levels that I just left it in the repo

ICubE01 commented on src/Task.java:
Task class has two parts. Static parts managing all tasks, instance parts of individual tasks.
It would be good to separate two parts to individual class.

woodenclock replied:
Will separate them out, I thought of Task as 1 entiry, needing both the task manager and task .
so you are suggesting a task class and a taskmanager class rit?

ICubE01 replied:
It's up to you, but yes, I suggest to do it.
Also, if you don't mind, trying to reduce similar codes will make your code more clear!

ICubE01 reviewed (commented):
I recommend you to divide your Task class into two parts; task manager and task itself.
You can use child class and override methods. (Todo, Deadline, Event)
Also, please remove dummy code.

woodenclock committed:
Update Elvis.java such that the Task and TaskManager are now separate objects

JoanneJo commented on src/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+public class Deadline extends Task {
+    private final char taskType = 'D';
```
Should the variable name be in all uppercase if it is a constant value? I noticed this constant in another class too.

JoanneJo commented on src/Elvis.java:
```diff
@@ -0,0 +1,58 @@
+            "       \\:::\\____\\                 \\:::\\____\\                                  \\:::\\____\\                 \\::::/    /       \n" +
+            "        \\::/    /                  \\::/    /                                   \\::/    /                  \\::/    /        \n" +
+            "         \\/____/                    \\/____/                                     \\/____/                    \\/____/         \n" +
+            "                                                                                                                                \n";
```
Perhaps can wrap the lines to keep it at less than 120 characters?

JoanneJo commented on src/Task.java:
```diff
@@ -0,0 +1,46 @@
+
+    //Setting the status of the isDone
+    public void setStatus(boolean truthValue) {
+        this.isDone = truthValue;
```
Would it be better to rename the boolean variable to sound like a boolean?

JoanneJo commented on src/TaskManager.java:
```diff
@@ -0,0 +1,148 @@
+                insertEvent(inputBuffer);
+            } else {
+                System.out.println("Invalid input!");
+            }
```
I like how neat the if-else statement is!

JoanneJo commented on src/TaskManager.java:
```diff
@@ -0,0 +1,148 @@
+        System.out.println("Got it. I've added this task:");
+        System.out.println("[" + 'T' + "]" +
+                "[" + list[listCount].getStatus() + "] " + list[listCount].getDescription());
+        System.out.println("Now you have " + (listCount+1) + " task(s) in the list.");
```
I think it is suggested to have white spaces between the operators? I noticed this is some other lines too.
```suggestion
        System.out.println("Now you have " + (listCount + 1) + " task(s) in the list.");
```

JoanneJo reviewed (commented):
The code looks great overall with minor issues

woodenclock committed:
Update Elvis.java separate dummy code

woodenclock committed:
Create ip

woodenclock committed:
Create Level-5 branching for addition of exception

woodenclock committed:
Merge branch 'branch-Level-5'

woodenclock committed:
Create branch for packaging.
Changed Dummy to dummy

woodenclock committed:
Update packaging

woodenclock committed:
Update packaging
Changed Dummy to dummy

woodenclock committed:
Add Help.java
Create Miscellaneous.java for storing basic functionalities

woodenclock committed:
Update TaskManager.java

woodenclock committed:
Merge branch 'branch-A-Packages'

# Conflicts:
#	src/elvis/TaskManager.java

woodenclock committed:
Update Elvis.java

woodenclock committed:
Update Elvis.java fixed some bugs

woodenclock committed:
Delete src/Dummy directory

woodenclock committed:
Add dummy

Musfirahe0556596 commented on src/dummy/List.java:
```diff
@@ -0,0 +1,42 @@
+import elvis.SystemOperation;
+import java.util.Scanner;
+
+//This was for WEEK 1, Code is not used
```
Remember to remove such comments!

Musfirahe0556596 commented on src/dummy/List.java:
```diff
@@ -0,0 +1,42 @@
+//This was for WEEK 1, Code is not used
+public class List {
+    public static void makeList() {
+        String[] list = new String[100];
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/elvis/Elvis.java:
```diff
@@ -0,0 +1,11 @@
+public class Elvis {
+
+    public static void main(String[] args) {
+        //Functionalities of ChatBot
+        //Echo.shoutEcho(); (WEEK 1)
+        //List.makeList();  (WEEK 2)
```
Remember to remove commented code!

Musfirahe0556596 commented on src/elvis/SystemOperation.java:
```diff
@@ -0,0 +1,53 @@
+                    "                                                                                                                                \n";
+
+    public static void printHorizontalLines() {
+        for (int i = 0; i < 60; i++) {
```
Try to avoid magic numbers, can make it a named constant

Musfirahe0556596 commented on src/elvis/TaskManager.java:
```diff
@@ -0,0 +1,218 @@
+package elvis;
+
+import elvis.exception.*;
```
Imported classes should always be listed explicitly (avoid *)

Musfirahe0556596 commented on src/elvis/TaskManager.java:
```diff
@@ -0,0 +1,218 @@
+public class TaskManager {
+    public static Scanner in = new Scanner(System.in);  //Scanner for Input
+    private static int listCount = 0;                   //Counter for the list filled up
+    private static final Task[] list = new Task[100];   //Keeps track of all Task Instances made
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 reviewed (commented):
Overall looks good but some coding standard and quality to follow!

Musfirahe0556596 commented on src/elvis/Help.java:
```diff
@@ -0,0 +1,46 @@
+
+//This is the help page to be displayed
+public class Help {
+    public static void help() {
```
You could rename this method better as verbs

woodenclock committed:
Improve code quality and coding standards

woodenclock committed:
Update README

woodenclock committed:
Update README

woodenclock committed:
Create branch for Level 6
Add new exception for empty delete
Create delete funtion

woodenclock committed:
Make small changes in the order of some functions

woodenclock committed:
Update branch for Level 6

woodenclock committed:
Update branch-Level-6

woodenclock committed:
Update for branch-Level-6

woodenclock committed:
Create branch-Level-7

woodenclock committed:
Update master

woodenclock committed:
Update for branch-Level-6

woodenclock committed:
Update branch Level 7

woodenclock committed:
Update input.txt and output.txt

woodenclock committed:
Update branch Level 7

woodenclock committed:
Update branch Level 7

woodenclock committed:
Update branch Level 7

woodenclock committed:
Merge branch 'branch-Level-6'

# Conflicts:
#	src/elvis/TaskManager.java

woodenclock committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/dummy/Echo.java
#	src/dummy/List.java
#	src/elvis/Elvis.java
#	src/elvis/TaskManager.java
#	text-ui-test/EXPECTED.TXT

woodenclock committed:
Delete dummy files

woodenclock committed:
Delete CONTRIBUTORS.md

woodenclock committed:
Update Elvis.java extract out classes

woodenclock committed:
Update main

woodenclock committed:
Update for extracting out the commands

woodenclock committed:
Update to fix bugs for delete command

woodenclock committed:
Branch for date and time addition

woodenclock committed:
Update final for date and time

woodenclock committed:
Update dummy for master branch

woodenclock committed:
Update Elvis for find function

woodenclock committed:
Merge pull request #1 from woodenclock/branch-Level-8

PR for branch level 8

woodenclock committed:
Merge branch 'master' into branch-Level-9

woodenclock committed:
Merge pull request #2 from woodenclock/branch-Level-9

PR for branch level 9

woodenclock committed:
Update Elvis.java remove redundant lines

woodenclock committed:
Update Elvis for JavaDoc

woodenclock committed:
Update README

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Delete tasks.txt

woodenclock committed:
Update DateTimeHandler to take care of errors

woodenclock committed:
Branch for A-JavaDoc

woodenclock committed:
Update Elvis and Ui for JavaDoc

woodenclock committed:
Update BootUpShutDown for JavaDoc

woodenclock committed:
Update gitignore

woodenclock committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/elvis/Elvis.java
#	src/elvis/operation/BootUpShutDown.java
#	src/elvis/operation/Ui.java

woodenclock committed:
Update Elvis for JavaDoc

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update .gitignore

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update Elvis for release V1.2

woodenclock committed:
Merge remote-tracking branch 'origin/master'

woodenclock committed:
Update Help

woodenclock committed:
Update README.md

woodenclock committed:
Update DateTimeHandler for more robust error handle
eg event go for the workshop /from 2023-6-29 1700 /to 58/06/2023 1900
where 58 is not valid

woodenclock committed:
Merge remote-tracking branch 'origin/master'

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update Help page

woodenclock committed:
Update for release

woodenclock committed:
Update README.md

woodenclock committed:
Update miscellaneous items

woodenclock committed:
Merge remote-tracking branch 'origin/master'

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update README.md

woodenclock committed:
Update gitignore

woodenclock committed:
Fix bugs found in peer reviews

woodenclock committed:
Merge remote-tracking branch 'origin/master'

woodenclock committed:
Update README.md
